### PR TITLE
Next Link is named correctly and no longer duplicated

### PR DESCRIPTION
**Pull request by mcardosos**
Fixes https://github.com/Azure/autorest/issues/2219
PTAL
@jhendrixMSFT @marstr

### CODE REVIEW

File: src/Model/CompositeTypeGo.cs [hunk 1]
```diff
@@ -226,35 +226,6 @@ public string Fields()
                 indented.Append(((CompositeTypeGo)BaseModelType).Fields());
             }
 
-            // If the type is a paged model type, ensure the nextLink field exists
-            // Note: Inject the field into a copy of the property list so as to not pollute the original list
-            if (!string.IsNullOrEmpty(NextLink))
```
Comment (jhendrixMSFT) [inline src/Model/CompositeTypeGo.cs]: If NextLink is no longer required please remove it.

Comment (mcardosos) [reply src/Model/CompositeTypeGo.cs]: It is still used on the template for paged methods convinience.

File: src/Model/MethodGo.cs [hunk 2]
```diff
@@ -557,27 +557,27 @@ public string NextLink
...
-                        var nextLinkName = (string)pageableExtension["nextLinkName"];
-                        if (!string.IsNullOrEmpty(nextLinkName))
+                        var nextLink = (string)pageableExtension["nextLinkName"];
+                        if (!string.IsNullOrEmpty(nextLink))
                         {
-                            nextLink = CodeNamerGo.PascalCaseWithoutChar(nextLinkName, '.');
+                            var nextLinkProperty = CodeNamerGo.Instance.GetPropertyName(CodeNamerGo.PascalCaseWithoutChar(nextLink, '.'));
+                            if ((ReturnType.Body as CompositeTypeGo).Properties.Any(p => p.Name.EqualsIgnoreCase(nextLinkProperty)))
+                            {
+                                (ReturnType.Body as CompositeTypeGo).Properties.First(p => p.Name.EqualsIgnoreCase(nextLinkProperty)).Name = nextLinkProperty;
```
Comment (jhendrixMSFT) [inline src/Model/MethodGo.cs]: It seems dubious that a getter is mutating state like this.  Shouldn't the name have been fixed up during transformation?

Comment (mcardosos) [reply src/Model/MethodGo.cs]: Fixed

**Comment by olydis** [comment]
# consider this as passing

Retriggered locally, went through. Sorry about the servers failing, we're fine tuning auto-publishing on PR merge right now...

**Comment by olydis** [comment]
# publish job

**Pull request merged**